boards/samd21-xpro/include/periph_conf.h

```diff
@@ -0,0 +1,277 @@
+
+#define UART_NUMOF          (sizeof(uart_config) / sizeof(uart_config[0]))
+/** @} */
+
+/**
+ * @name PWM configuration
+ * @{
+ */
+
+#define PWM_0_EN            1
+#define PWM_1_EN            0
+#define PWM_2_EN            0
+#define PWM_MAX_CHANNELS    2
+/* for compatibility with test application */
+#define PWM_0_CHANNELS      PWM_MAX_CHANNELS
+#define PWM_1_CHANNELS      PWM_MAX_CHANNELS
+#define PWM_2_CHANNELS      PWM_MAX_CHANNELS
+
+/* PWM device configuration */
+static const pwm_conf_t pwm_config[] = {
+#if PWM_0_EN
+    {TCC2, {
+        /* GPIO pin, MUX value, TCC channel */
+        { GPIO_PIN(PA, 12), GPIO_MUX_E, 0 },
+        { GPIO_PIN(PA, 13), GPIO_MUX_E, 1 },
+    }},
+#endif
+#if PWM_1_EN
+    {TC4, {
+        /* GPIO pin, MUX value, TCC channel */
+        { GPIO_PIN(PB, 12), GPIO_MUX_E, 0 },
+        { GPIO_PIN(PB, 13), GPIO_MUX_E, 1 },
+    }}
+#endif
+#if PWM_2_EN
+    {TC6, {
+        /* GPIO pin, MUX value, TCC channel */
+        { GPIO_PIN(PB, 02), GPIO_MUX_E, 0 },
+        { GPIO_PIN(PB, 03), GPIO_MUX_E, 1 },
+    }}
+#endif
+};
+
+/* number of devices that are actually defined */
+#define PWM_NUMOF           (3U)
+/** @} */
+
+/**
+ * @name SPI configuration
+ * @{
+ */
+static const spi_conf_t spi_config[] = {
+
+    {
+        .dev      = &SERCOM0->SPI,
+        .miso_pin = GPIO_PIN(PA, 04),
+        .mosi_pin = GPIO_PIN(PA, 06),
+        .clk_pin  = GPIO_PIN(PA, 07),
+        .miso_mux = GPIO_MUX_D,
+        .mosi_mux = GPIO_MUX_D,
+        .clk_mux  = GPIO_MUX_D,
+        .miso_pad = SPI_PAD_MISO_0,
+        .mosi_pad = SPI_PAD_MOSI_2_SCK_3
+    },
+    {
+        .dev      = &SERCOM1->SPI,
+        .miso_pin = GPIO_PIN(PA, 16),
+        .mosi_pin = GPIO_PIN(PA, 18),
+        .clk_pin  = GPIO_PIN(PA, 19),
+        .miso_mux = GPIO_MUX_C,
+        .mosi_mux = GPIO_MUX_C,
+        .clk_mux  = GPIO_MUX_C,
+        .miso_pad = SPI_PAD_MISO_0,
+        .mosi_pad = SPI_PAD_MOSI_2_SCK_3
+    },
+    {
+        .dev      = &SERCOM5->SPI,
+        .miso_pin = GPIO_PIN(PB, 16),
+        .mosi_pin = GPIO_PIN(PB, 22),
+        .clk_pin  = GPIO_PIN(PB, 23),
+        .miso_mux = GPIO_MUX_C,
+        .mosi_mux = GPIO_MUX_C,
+        .clk_mux  = GPIO_MUX_C,
+        .miso_pad = SPI_PAD_MISO_0,
+        .mosi_pad = SPI_PAD_MOSI_2_SCK_3
+    }
+};
+
+#define SPI_NUMOF           (sizeof(spi_config) / sizeof(spi_config[0]))
+/** @} */
+
+/**
+ * @name I2C configuration
+ * @{
+ */
+#define I2C_NUMOF          (1U)
+#define I2C_0_EN            1
+#define I2C_1_EN            0
+#define I2C_2_EN            0
+#define I2C_3_EN            0
+#define I2C_IRQ_PRIO        1
+
+#define I2C_0_DEV           SERCOM2->I2CM
+#define I2C_0_IRQ           SERCOM2_IRQn
+#define I2C_0_ISR           isr_sercom2
+/* I2C 0 GCLK */
+#define I2C_0_GCLK_ID       SERCOM2_GCLK_ID_CORE
+#define I2C_0_GCLK_ID_SLOW  SERCOM2_GCLK_ID_SLOW
+/* I2C 0 pin configuration */
+#define I2C_0_SDA           GPIO_PIN(PA, 08)
+#define I2C_0_SCL           GPIO_PIN(PA, 09)
```

Comment:
I get a compilation error here complaining about the leading 0 on arm-none-eabi-gcc 5.4.1.